### PR TITLE
[KOGITO-550] Fix error swallow issues of the reconcile loop

### PR DESCRIPTION
**Pull request by xiezhang7**
Description: fix the reconcile loop to exit when any error is encountered.

Jira: https://issues.jboss.org/browse/KOGITO-550

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: pkg/controller/kogitoapp/kogitoapp_controller_test.go [hunk 1]
```diff
@@ -191,21 +212,26 @@ func TestKogitoAppWithResource(t *testing.T) {
...
-		&v1alpha1.KogitoAppList{})
-	s.AddKnownTypes(appsv1.GroupVersion,
-		&appsv1.DeploymentConfig{},
-		&appsv1.DeploymentConfigList{})
+	s.AddKnownTypes(v1alpha1.SchemeGroupVersion, kogitoapp, &v1alpha1.KogitoAppList{})
+	s.AddKnownTypes(appsv1.GroupVersion, &appsv1.DeploymentConfig{}, &appsv1.DeploymentConfigList{})
 	s.AddKnownTypes(buildv1.GroupVersion, &buildv1.BuildConfig{}, &buildv1.BuildConfigList{})
 	s.AddKnownTypes(routev1.GroupVersion, &routev1.Route{}, &routev1.RouteList{})
 	s.AddKnownTypes(imgv1.GroupVersion, &imgv1.ImageStreamTag{}, &imgv1.ImageStream{}, &imgv1.ImageStreamList{})
 	s.AddKnownTypes(monv1.SchemeGroupVersion, &monv1.ServiceMonitor{}, &monv1.ServiceMonitorList{})
```
Comment (ricardozanini) [inline pkg/controller/kogitoapp/kogitoapp_controller_test.go]: Please add those types to the `meta.GetRegisteredSchema()` function instead.

File: pkg/controller/kogitoapp/kogitoapp_controller_test.go [hunk 2]
```diff
@@ -272,3 +298,238 @@ func TestKogitoAppWithResource(t *testing.T) {
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kogitoapp.Name,
+			Namespace: kogitoapp.Namespace,
+		},
+		Spec: routev1.RouteSpec{
+			Host: "localhost",
+		},
+	}
+
+	clientfake := &kogitoclient.Client{
```
Comment (ricardozanini) [inline pkg/controller/kogitoapp/kogitoapp_controller_test.go]: Please use the CreateFakeClient method from the `pkg/test` instead.
EDIT: I refactored it to include Prometheus and Discovery in https://github.com/kiegroup/kogito-cloud-operator/pull/100. You can rebase after we merge it.

File: pkg/controller/kogitoapp/kogitoapp_controller_test.go [hunk 3]
```diff
@@ -272,3 +298,238 @@ func TestKogitoAppWithResource(t *testing.T) {
...
+				kogitoAppResources,
+				&status.UpdateResourcesResult{},
+			},
+			&reconcile.Result{Requeue: true},
+			nil,
+		},
+		{
+			"RequeueAfter",
+			fields{
+				&kogitoclient.Client{
```
Comment (ricardozanini) [inline pkg/controller/kogitoapp/kogitoapp_controller_test.go]: Same here regarding the client

File: pkg/controller/kogitoapp/kogitoapp_controller_test.go (hunk 1 above)
Comment (xiezhang7) [reply pkg/controller/kogitoapp/kogitoapp_controller_test.go]: corrected

File: pkg/controller/kogitoapp/kogitoapp_controller_test.go (hunk 2 above)
Comment (xiezhang7) [reply pkg/controller/kogitoapp/kogitoapp_controller_test.go]: corrected

File: pkg/controller/kogitoapp/kogitoapp_controller_test.go (hunk 3 above)
Comment (xiezhang7) [reply pkg/controller/kogitoapp/kogitoapp_controller_test.go]: corrected

**Pull request merged**